### ISSUE TITLE
Sink: Kafka buffer.type= "disk" -> flush fixed at 100 messages

### ISSUE THREAD

**Issue by lzwaan** [vectordotdev/vector#10982]
### Issue
When using Sink: Kafka buffer.type= "disk", no matter what parameter i try to change, it will flush to Kafka Topic after exactly 100 messages. If the total amount of messages will be below 100 messages it will stays on disk forever without sending it to Kafka Topic

### Expected behaviour
To set a parameter to flush the remaining messages (20 messages in the example below) after like 5 seconds?

### Example:
send 30 messages  -> shown in console -> Kafka Topic keeps empty
send 30 messages  -> shown in console -> Kafka Topic keeps empty
send 30 messages  -> shown in console -> Kafka Topic keeps empty
send 30 messages  -> shown in console -> Kafka Topic show 100 messages (the other 20 messages not shown)
stop vector container -> Now Kafka Topic shows a total of 120 messages, which is correct

### My current setup in docker-compose:
```
vector:
    image: timberio/vector:0.19.0-debian
    container_name: vector
    hostname: vector
    ports:
      - "8686:8686"
    volumes:
      - ./scripts/vector/etc/radius/vector.toml:/etc/vector/vector.toml:ro
      - ./scripts/vector/logfiles/radius:/var/log/vector/logfiles
      - vector:/var/lib/vector
    environment:
      TZ: "${TZ:-Europe/Amsterdam}"
volumes:
  vector:
```

### vector.toml
```
# Global options
data_dir                                    = "/var/lib/vector"
log_schema.message_key                      = "message"
timezone                                    = "Europe/Amsterdam"

[api]
enabled                                     = false

[sources.logs]
  type                                      = "file"
  include                                   = [ "/var/log/vector/logfiles/*.json" ]

[transforms.transform_radius]
  type                                      = "remap"
  inputs                                    = [ "logs" ]
  source                                    = '''
      . = parse_json!(.message)
      if (exists(.time)) {
          .timestamp = to_unix_timestamp(parse_timestamp!(to_string!(.time), format: "%s%.f"), unit: "milliseconds")
      } else {
         .timestamp = to_unix_timestamp(now(), unit: "milliseconds")
      }
 '''

[sinks.out]
  type                                      = "console"
  inputs                                    = ["transform_radius"]
  encoding.codec                            = "json"
  encoding.timestamp_format                 = "unix"

[sinks.kafka_sink]
  type                                      = "kafka"
  inputs                                    = [ "transform_radius" ]
  bootstrap_servers                         = "kafka1:10091,kafka2:10092"

  buffer.type                               = "disk"
  buffer.max_size                           = 100000000  # 100 MB

  buffer.when_full                          = "block"
  librdkafka_options."message.timeout.ms"   = "0"
  librdkafka_options."acks"                 = "all"
  librdkafka_options."enable.idempotence"   = "true"
  librdkafka_options."log.connection.close" = "false"
  librdkafka_options."client.id"            = "vector-radius-ingress"
  topic                                     = "input..radius"
  compression                               = "snappy"
  encoding.codec                            = "json"
  encoding.timestamp_format                 = "unix"
```

**Comment by jeromekleinen** [comment]
I believe this is the same issue as https://github.com/vectordotdev/vector/issues/10806 which will be fixed in 0.19.1, which should be released next week.

**Comment by spencergilbert** [comment]
Yep, does sound like a duplicate of the linked issue. Using a memory buffer can temporarily resolve the issue, but the fix should be released this week.

Apologies for the trouble!

**Comment by jszwedko** [comment]
Indeed this was released as 0.19.1.
